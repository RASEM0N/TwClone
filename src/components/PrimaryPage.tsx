import React from 'react'
import { makeStyles, Typography, Button } from '@material-ui/core'
import TwitterIcon from '@material-ui/icons/Twitter'
import SearchIcon from '@material-ui/icons/Search'
import PeopleOutlineIcon from '@material-ui/icons/PeopleOutline'
import ModeCommentOutlinedIcon from '@material-ui/icons/ModeCommentOutlined'
import { useHistory } from 'react-router-dom'

const useStyles = makeStyles((theme) => ({
    wrapper: {
        display: 'flex',
        height: '100vh',
    },

    /* info side */
    infoSide: {
        flex: 0.5,
        backgroundColor: '#71C9F8',
        height: '100%',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'relative',
        overflow: 'hidden',

        '& > ul': {
            zIndex: 1,
            listStyle: 'none',
            padding: 0,
            margin: 0,
            width: 400,
        },

        '& h5': {
            display: 'flex',
            alignItems: 'center',
            fontSize: 20,
            color: 'white',
            fontWeight: 700,
        },

        '& li': {
            marginBottom: theme.spacing(4),
        },
    },
    infoSideIcon: {
        fontSize: 32,
        marginRight: 20,
    },
    infoIconTwitterBcg: {
        position: 'absolute',
        left: '50%',
        top: '53%',
        transform: 'translate(-50%, -50%)',
        width: '250%',
        height: '250%',
        zIndex: 0,
    },

    /* auth side */
    authSide: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        flex: 0.5,
    },
    authIconTwitter: {
        fontSize: 50,
    },
    authSideWrapper: {
        maxWidth: 380,
    },
    authSideTitle: {
        fontWeight: 600,
        fontSize: 32,
        marginBottom: 45,
        marginTop: 20,
    },
}))

const PrimaryPage: React.FunctionComponent = () => {
    const classes = useStyles()
    const history = useHistory()

    const toLogin = () => {
        history.push('/auth/login')
    }
    const toRegister = () => {
        history.push('/auth/register')
    }

    return (
        <div className={classes.wrapper}>
            <div className={classes.infoSide}>
                <TwitterIcon className={classes.infoIconTwitterBcg} color="primary" />
                <ul>
                    <li>
                        <Typography variant="h5">
                            {' '}
                            <SearchIcon className={classes.infoSideIcon} />
                            Читайте о том, что вам интересно
                        </Typography>
                    </li>
                    <li>
                        <Typography variant="h5">
                            <PeopleOutlineIcon className={classes.infoSideIcon} />
                            Узнайте о, чем говорят в мире{' '}
                        </Typography>
                    </li>
                    <li>
                        <Typography variant="h5">
                            <ModeCommentOutlinedIcon className={classes.infoSideIcon} />
                            Присоединяйтесь к общению
                        </Typography>
                    </li>
                </ul>
            </div>
            <section className={classes.authSide}>
                <div className={classes.authSideWrapper}>
                    <TwitterIcon className={classes.authIconTwitter} color="primary" />
                    <Typography className={classes.authSideTitle} variant="h4">
                        Узнайте, что происходит в мире сейчас
                    </Typography>
                    <Typography>
                        <b>Присоедийтесь к Твиттеру прямо сейчас!</b>
                    </Typography>
                    <Button
                        style={{
                            marginBottom: 20,
                        }}
                        variant="contained"
                        fullWidth
                        color="primary"
                        onClick={toRegister}
                    >
                        Зарегистрироваться
                    </Button>
                    <Button onClick={toLogin} variant="outlined" fullWidth color="primary">
                        Войти
                    </Button>
                </div>
            </section>
        </div>
    )
}

export default PrimaryPage
