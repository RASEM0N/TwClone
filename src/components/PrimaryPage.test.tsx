// @ts-ignore
import PrimaryPage from './PrimaryPage'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('PrimaryPage', () => {
    it('just render', () => {
        const Comp = <PrimaryPage />
        const component = shallow(Comp)
        expect(component.debug()).toMatchSnapshot()
    })
})
