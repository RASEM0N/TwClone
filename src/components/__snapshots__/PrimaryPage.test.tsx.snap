// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryPage just render 1`] = `
"<div className=\\"makeStyles-wrapper-1\\">
  <div className=\\"makeStyles-infoSide-2\\">
    <Memo(ForwardRef(TwitterIcon)) className=\\"makeStyles-infoIconTwitterBcg-4\\" color=\\"primary\\" />
    <ul>
      <li>
        <WithStyles(ForwardRef(Typography)) variant=\\"h5\\">
           
          <Memo(ForwardRef(SearchIcon)) className=\\"makeStyles-infoSideIcon-3\\" />
          Читайте о том, что вам интересно
        </WithStyles(ForwardRef(Typography))>
      </li>
      <li>
        <WithStyles(ForwardRef(Typography)) variant=\\"h5\\">
          <Memo(ForwardRef(PeopleOutlineIcon)) className=\\"makeStyles-infoSideIcon-3\\" />
          Узнайте о, чем говорят в мире
           
        </WithStyles(ForwardRef(Typography))>
      </li>
      <li>
        <WithStyles(ForwardRef(Typography)) variant=\\"h5\\">
          <Memo(ForwardRef(ModeCommentOutlinedIcon)) className=\\"makeStyles-infoSideIcon-3\\" />
          Присоединяйтесь к общению
        </WithStyles(ForwardRef(Typography))>
      </li>
    </ul>
  </div>
  <section className=\\"makeStyles-authSide-5\\">
    <div className=\\"makeStyles-authSideWrapper-7\\">
      <Memo(ForwardRef(TwitterIcon)) className=\\"makeStyles-authIconTwitter-6\\" color=\\"primary\\" />
      <WithStyles(ForwardRef(Typography)) className=\\"makeStyles-authSideTitle-8\\" variant=\\"h4\\">
        Узнайте, что происходит в мире сейчас
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))>
        <b>
          Присоедийтесь к Твиттеру прямо сейчас!
        </b>
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Button)) style={{...}} variant=\\"contained\\" fullWidth={true} color=\\"primary\\" onClick={[Function: toRegister]}>
        Зарегистрироваться
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button)) onClick={[Function: toLogin]} variant=\\"outlined\\" fullWidth={true} color=\\"primary\\">
        Войти
      </WithStyles(ForwardRef(Button))>
    </div>
  </section>
</div>"
`;
